src/shared/clients/codecatalystClient.ts

```diff
@@ -411,12 +413,45 @@ class CodeCatalystClientInternal {
 
     /**
      * Gets a flat list of all repos for the given CodeCatalyst user.
+     * @param internalOnly True if you want only internal repositories to be returned.
      */
     public listSourceRepositories(
-        request: CodeCatalyst.ListSourceRepositoriesRequest
+        request: CodeCatalyst.ListSourceRepositoriesRequest,
+        internalOnly: boolean = false
```

Comment:
This is the core of the change. When listing repositories the user has the option to indicate they only want internal repos.